cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/SchemaContainer.tsx

```diff
@@ -0,0 +1,216 @@
+      fontSize: '9px',
+      marginLeft: '5px',
+    },
+    // Table cell has grey border bottom by default, this causes issues when we
+    // changeborder to red for rows (cell overrides row color). So, we remove
+    // any border from table cells, add border to rows
+    tableRow: {
+      '& td': {
+        border: 0,
+      },
+      borderBottom: `1px solid ${theme.palette.grey[400]}`,
+    },
+    errorRow: {
+      // each errored cell would have red top border to combat row border from
+      // previous row
+      '& td': {
+        borderTop: `1px solid ${theme.palette.red[200]}`,
+      },
+      border: `1px solid ${theme.palette.red[200]}`,
+    },
+  };
+};
+
+interface ISchemaContainerProps extends WithStyles<typeof styles> {
+  stage: IParsedInputSchema;
+  onExpandClick: (stage: IParsedInputSchema) => void;
+  onSchemaChange: (stage: IParsedInputSchema) => void;
+  aliases: object;
+  errorCount: number;
+}
+
+interface ISchemaContainerState {
+  selectFields: string;
+  menuAnchor: HTMLButtonElement;
+}
+
+class SchemaContainer extends React.Component<ISchemaContainerProps, ISchemaContainerState> {
+  public state = {
+    selectFields: 'All',
+    menuAnchor: null,
+  };
+
+  private tableSelectChange = (selectFields) => {
+    this.setState({ selectFields }, () => {
+      const newSchema = this.props.stage.schema.map((field) => {
+        return {
+          ...field,
+          selected: this.state.selectFields === 'All' ? true : false,
+        };
+      });
+      this.props.onSchemaChange(this.constructNewStage(newSchema));
+    });
+  };
+
+  private handleMenuClose = (value) => {
+    this.setState({ menuAnchor: null }, () => {
+      if (value) {
+        this.tableSelectChange(value);
+      }
+    });
+  };
+
+  private handleMenuOpen = (event) => {
+    this.setState({ menuAnchor: event.currentTarget });
+  };
+
+  private constructNewStage = (newSchema) => {
+    return {
+      ...this.props.stage,
+      schema: newSchema,
+    };
+  };
+
+  private toggleExpansionPanel = () => {
+    this.props.onExpandClick(this.props.stage);
+  };
+
+  private updateField = (newField) => {
+    if (this.props.stage && this.props.stage.schema) {
+      const newSchema = this.props.stage.schema.map((field) => {
+        if (field.name === newField.name) {
+          return newField;
+        }
+        return field;
+      });
+      this.props.onSchemaChange(this.constructNewStage(newSchema));
+    }
+  };
+
+  public render() {
+    const { classes } = this.props;
+    return (
+      <ExpansionPanel
+        className={classes.schemaContainer}
+        expanded={this.props.stage.expanded}
+        onChange={this.toggleExpansionPanel}
+      >
+        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
+          <Typography variant="h6">{this.props.stage.name}</Typography>
+          <If condition={Boolean(this.props.errorCount)}>
+            <span className={classnames('badge', classes.badgeDanger)}>
```

Comment:
there should be an equivalent badge component from material-ui?

Reply:
@elfenheart material badges are a bit different, they appear like super script: https://material-ui.com/components/badges/ 
If we decide to use this design, let's make these changes when we revisit the widgets for styling?